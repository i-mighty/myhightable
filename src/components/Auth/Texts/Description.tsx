import styled from 'styled-components/native';
import React from 'react';

interface DescriptionProps {
  color?: string;
  text: string;
  testID?: string;
  fontSize?: number;
}

const StyledDescription = styled.Text<
  Pick<DescriptionProps, 'color' | 'fontSize'>
>`
  font-family: Roboto;
  font-style: normal;
  font-weight: normal;
  font-size: ${({ fontSize }) => fontSize || 16}px;
  line-height: 21px;
  text-align: center;
  color: ${({ color, theme }) => color || theme.vars.slate_grey};
`;

const Description: React.FC<DescriptionProps> = ({
  text,
  testID,
  color,
  fontSize,
}) => {
  return (
    <StyledDescription color={color} testID={testID} fontSize={fontSize}>
      {text}
    </StyledDescription>
  );
};

export default Description;
