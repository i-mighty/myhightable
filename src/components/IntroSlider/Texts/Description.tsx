import styled from 'styled-components/native';
import React from 'react';

interface DescriptionProps {
  color?: string;
  text: string;
  testID?: string;
}

const StyledDescription = styled.Text<Pick<DescriptionProps, 'color'>>`
  font-family: Roboto;
  font-style: normal;
  font-weight: normal;
  font-size: 18px;
  line-height: 21px;
  text-align: center;
  color: ${({ color, theme }) => color || theme.vars.white};
  margin-top: 10px;
`;

const Description: React.FC<DescriptionProps> = ({ text, testID, color }) => {
  return (
    <StyledDescription color={color} testID={testID}>
      {text}
    </StyledDescription>
  );
};

export default Description;
