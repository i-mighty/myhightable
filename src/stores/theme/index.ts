interface Theme {
  vars: {
    [key: string]: string;
  };
  dimens?: {
    [key: string]: string;
  };
}

const theme: Theme = {
  vars: {
    orange: '#F89505',
    dark: '#000000',
    black: '#333333',
    black_alt: '#4F4F4F',
    slate_grey: '#828282',
    grey: '#BDBDBD',
    grey_alt: '#C4C4C4',
    white: '#FFFFFF',
    white_alt: '#F2F2F2',
  },
};

export default theme;
