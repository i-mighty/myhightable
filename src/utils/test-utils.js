import { ThemeProvider } from 'styled-components/native';
import React from 'react';
import { render as baseRender } from '@testing-library/react-native';
import theme from '@src/stores/theme';

export const renderWithProps = (Component, props) => {
  baseRender(
    <ThemeProvider theme={theme}>
      <Component {...props} />
    </ThemeProvider>,
  );
};
