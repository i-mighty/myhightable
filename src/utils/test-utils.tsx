import { ThemeProvider } from 'styled-components/native';
import React, { ReactElement } from 'react';
import { render as baseRender } from '@testing-library/react-native';
import theme from '@src/stores/theme';

export const renderWithProps = (
  Component: React.ElementType,
  props: Object,
) => {
  return baseRender(
    <ThemeProvider theme={theme}>
      <Component {...props} />
    </ThemeProvider>,
  );
};
