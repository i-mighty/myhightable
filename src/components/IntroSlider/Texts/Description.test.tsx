import Description from './Description';
import theme from '@src/stores/theme';
import { renderWithProps } from '@src/utils/test-utils';

const defaultProps = { testID: 'title', text: 'HighTable' };
const colouredProps = { ...defaultProps, color: theme.vars.orange };

describe('Description', () => {
  it('renders successfully', () => {
    const { getByTestId, baseElement } = renderWithProps(
      Description,
      defaultProps,
    );
    expect(getByTestId(defaultProps.testID).props.children).toEqual(
      defaultProps.text,
    );
    expect(baseElement).toMatchSnapshot();
  });
  it('renders with the right color', () => {
    const { getByTestId, baseElement } = renderWithProps(
      Description,
      colouredProps,
    );
    const textComponent = getByTestId(colouredProps.testID);
    expect(textComponent.props.children).toEqual(colouredProps.text);
    expect(textComponent).toHaveStyleRule('color', colouredProps.color);
    expect(baseElement).toMatchSnapshot();
  });
});
