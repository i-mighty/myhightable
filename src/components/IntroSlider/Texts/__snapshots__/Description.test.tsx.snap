// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description renders successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Roboto",
            "fontSize": 18,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "lineHeight": 21,
            "marginTop": 10,
            "textAlign": "center",
          },
        ]
      }
      testID="title"
    >
      HighTable
    </Text>
  </View>
</View>
`;

exports[`Description renders with the right color 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      color="#F89505"
      style={
        Array [
          Object {
            "color": "#F89505",
            "fontFamily": "Roboto",
            "fontSize": 18,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "lineHeight": 21,
            "marginTop": 10,
            "textAlign": "center",
          },
        ]
      }
      testID="title"
    >
      HighTable
    </Text>
  </View>
</View>
`;
